src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,182 @@
+            description: [
+                'Sets the vertical domain of this `parcats` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        editType: 'calc'
+    },
+
+    tooltip: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'info',
+        editType: 'plot',
+        description: 'Shows a tooltip when hover mode is `category` or `color`.'
+    },
+
+    hovermode: {
+        valType: 'enumerated',
+        values: ['none', 'category', 'color'],
+        dflt: 'category',
+        role: 'info',
+        editType: 'plot',
+        description: 'Sets the hover mode of the parcats diagram'
+    },
+    bundlecolors: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'info',
+        editType: 'plot',
+        description: 'Sort paths so that like colors are bundled together'
+    },
+    sortpaths: {
+        valType: 'enumerated',
+        values: ['forward', 'backward'],
+        dflt: 'forward',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'If `forward` then sort paths based on dimensions from left to right.',
+            'If `backward` sort based on dimensions from right to left.'
+        ].join(' ')
+    },
+    // labelfont: fontAttrs({
+    //     editType: 'calc',
+    //     description: 'Sets the font for the `dimension` labels.'
+    // }),
+    //
+    // catfont: fontAttrs({
+    //     editType: 'calc',
+    //     description: 'Sets the font for the `category` labels.'
+    // }),
+
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+        label: {
+            valType: 'string',
+            role: 'info',
+            editType: 'calc',
+            description: 'The shown name of the dimension.'
+        },
+        catDisplayInds: {
+            valType: 'data_array',
+            role: 'info',
+            editType: 'calc',
+            dflt: [],
+            description: [
+                ''
+            ].join(' ')
+        },
+        catValues: {
+            valType: 'data_array',
+            role: 'info',
+            editType: 'calc',
+            dflt: [],
+            description: [
+                ''
+            ].join(' ')
+        },
+        catLabels: {
+            valType: 'data_array',
+            role: 'info',
+            editType: 'calc',
+            dflt: [],
+            description: [
+                ''
+            ].join(' ')
+        },
+        values: {
+            valType: 'data_array',
+            role: 'info',
+            dflt: [],
+            editType: 'calc',
+            description: [
+                'Dimension values. `values[n]` represents the category value of the `n`th point in the dataset,',
+                'therefore the `values` vector for all dimensions must be the same (longer vectors',
+                'will be truncated). Each value must an element of `catValues`.'
+            ].join(' ')
+        },
+        displayInd: {
```

Comment:
Again I'd spell out `displayindex`. Also reminds me we need a `visible` attribute per dimension.

Reply:
Sure. Do you prefer all lowercase for the internal JavaScript object properties as well? Right now all of these are in camelCase, but I can change that.

Reply:
Just use lowercase for attribute names in the figure JSON (data and layout). In the code, camelCase is preferred.

Reply:
Done in https://github.com/jonmmease/plotly.js/pull/1/commits/a34dafaf6f344e4b3936626b1c144ab8969877da

Reply:
Visible done in 2c14168 and c8e3cc9